File: backend/SindhDormitory.Domain/Entities/HostelAmenity.cs
// =============================================================================
// Domain/Entities/HostelAmenity.cs
// =============================================================================
namespace SindhDormitory.Domain.Entities;

public class HostelAmenity : BaseEntity
{
    public int     AmenityId    { get; set; }
    public int     HostelId     { get; set; }
    public string  AmenityName  { get; set; } = string.Empty;
    public string? Description  { get; set; }

    // Navigation
    public Hostel Hostel { get; set; } = null!;
}

File: backend-node/src/entities/HostelAmenity.ts
import {
  Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn,
  CreateDateColumn, UpdateDateColumn
} from 'typeorm';
import { Hostel } from './Hostel';

@Entity('HostelAmenities')
export class HostelAmenity {
  @PrimaryGeneratedColumn({ name: 'AmenityId' })
  amenityId!: number;

  @Column({ name: 'HostelId', type: 'int' })
  hostelId!: number;

  @Column({ name: 'AmenityName', type: 'varchar', length: 100 })
  amenityName!: string;

  @CreateDateColumn({ name: 'CreatedAt' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'UpdatedAt' })
  updatedAt!: Date;

  @ManyToOne(() => Hostel, (h) => h.amenities, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'HostelId' })
  hostel!: Hostel;
}
